rfc/rfc-62/rfc-62.md

```diff
@@ -0,0 +1,443 @@
+                            "prevCommit":"null",
+                            "numWrites":123352,
+                            "numDeletes":0,
+                            "numUpdateWrites":0,
+                            "numInserts":123352,
+                            "totalWriteBytes":4675371,
+                            "totalWriteErrors":0,
+                            "tempPath":null,
+                            "partitionPath":"20210623/0/20210825",
+                            "totalLogRecords":0,
+                            "totalLogFilesCompacted":0,
+                            "totalLogSizeCompacted":0,
+                            "totalUpdatedRecordsCompacted":0,
+                            "totalLogBlocks":0,
+                            "totalCorruptLogBlock":0,
+                            "totalRollbackBlocks":0,
+                            "fileSizeInBytes":4675371,
+                            "minEventTime":null,
+                            "maxEventTime":null
+                        }
+                    ]
+                },
+                "compacted":false,
+                "extraMetadata":{
+                    "schema":"xxxx"
+                },
+                "operationType":"UPSERT",
+                "totalRecordsDeleted":0,
+                "totalLogFilesSize":0,
+                "totalScanTime":0,
+                "totalCreateTime":21051,
+                "totalUpsertTime":0,
+                "minAndMaxEventTime":{
+                    "Optional.empty":{
+                        "val":null,
+                        "present":false
+                    }
+                },
+                "writePartitionPaths":[
+                    "20210623/0/20210825"
+                ],
+                "fileIdAndRelativePaths":{
+                    "c144908e-ca7d-401f-be1c-613de98d96a3-0":"20210623/0/20210825/c144908e-ca7d-401f-be1c-613de98d96a3-0_3-33-2009_20220818134233973.parquet"
+                },
+                "totalLogRecordsCompacted":0,
+                "totalLogFilesCompacted":0,
+                "totalCompactedRecordsUpdated":0
+            }
+        }
+    ]
+}
+```
+
+### Engine Runtime Information
+
+Spark allow users to get metrics through api as a json same as spark ui:
+> In addition to viewing the metrics in the UI, they are also available as JSON. This gives developers an easy way to create new visualizations and monitoring tools for Spark. 
+> The JSON is available for both running applications, and in the history server. The endpoints are mounted at /api/v1. 
+> For example, for the history server, they would typically be accessible at http://<server-url>:18080/api/v1, and for a running application, at http://localhost:4040/api/v1.
+
+So that we can collect any spark runtime information we want. For example collect all the stages and collect the corresponding execution time
```

Comment:
Though it may not related to engine runtime information, could we keep track of instants produced by the current engine and record them somewhere? 

Because in many cases, users run multiple clients without OCC turned on. When error occurred, the exception message cannot reveal the root cause (i.e., multi-client issue).

Reply:
Sure for each commit, this Diagnostic Reporter will generate report json with runtime info at `.hoodie/report/instant-time/`